### PR TITLE
feat(datasets): implement removeAllAccessPermissions and update removeDataSet

### PR DESCRIPTION
**Pull request by DaveMadsen**
Issue #, if available:

Description of changes:

Checklist:

* [X] Have you successfully deployed to an AWS account with your changes?
* [X] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-reference/src/services/dataSetService.ts [hunk 1]
```diff
@@ -102,7 +102,10 @@ export class DataSetService implements DataSetPlugin {
   }
 
   public async removeAllAccessPermissions(datasetId: string): Promise<PermissionsResponse> {
-    const response = await this._dataSetsAuthService.removeAllAccessPermissions(datasetId);
+    const response = await this._dataSetsAuthService.removeAllAccessPermissions(datasetId, {
```
Comment (aws-tyler) [inline solutions/swb-reference/src/services/dataSetService.ts:105]: is this just a placeholder?

File: workbench-core/datasets/src/dataSetsAuthorizationPlugin.ts [hunk 2]
```diff
@@ -55,8 +55,11 @@ export interface DataSetsAuthorizationPlugin {
...
    * Remove all permssions from a given dataset.
    * @param datasetId - the ID of the datset for which permissions are to be removed.
+   * @param authenticatedUser - an object representing the user making the request.
    * @returns a {@link PermissionsResponse} indicating the permissions removed.
-   * IMPORTANT: If the returned object has a non-null `pageToken`, there are more permissions to delete.
    */
-  removeAllAccessPermissions(datasetId: string): Promise<PermissionsResponse>;
+  removeAllAccessPermissions(
+    datasetId: string,
+    authenticatedUser: { id: string; roles: string[] }
```
Comment (aws-tyler) [inline workbench-core/datasets/src/dataSetsAuthorizationPlugin.ts:63]: I'm seeing this definition in multiple places, might be worth making a type for it.

**Review by aws-tyler — APPROVED** [review]

File: solutions/swb-reference/src/services/dataSetService.ts (hunk 1 above)
Comment (DaveMadsen) [reply solutions/swb-reference/src/services/dataSetService.ts:105]: oh, the authenticatedUser? Yes! `res.locals.user` should be passed from the route handler.

File: workbench-core/datasets/src/dataSetsAuthorizationPlugin.ts (hunk 2 above)
Comment (DaveMadsen) [reply workbench-core/datasets/src/dataSetsAuthorizationPlugin.ts:63]: Yes! At some point, we should refactor our reference implementation plugins into separate packages so our packages are not bound to each other. Ideally that change should affect only the authz plugin, and not the interface. If I pull `AuthenticatedUser` from `workbench-core-authorization` now, I will have to make the change to the interface later to tease the packages apart. I favor not setting up future interface changes even if they seem trivial.

Comment (htztomic) [reply workbench-core/datasets/src/dataSetsAuthorizationPlugin.ts:63]: It will be naturally coupled by the way we depend on authenticatedUser but I agree

**Review by htztomic — APPROVED** [review]
LGTM

**Pull request merged**